application/lib/view/auth_page.dart

```diff
@@ -37,20 +61,24 @@ class AuthPage extends StatelessWidget {
     );
   }
 
-  Widget buildIdentifiersInput() {
-    return const Column(
+  Widget buildIdentifiersInput(BuildContext context) {
+    return Column(
       children: [
         AVAITextInput(
           label: 'E-mail institucional',
           placeholder: 'nome@ic.ufrj.br',
           allowNumbers: true,
           allowSpaces: false,
+          onChanged: (value) =>
+              Provider.of<AuthModel>(context, listen: false).setEmail(value),
         ),
-        SizedBox(height: 24),
+        const SizedBox(height: 24),
         AVAITextInput(
           label: 'Senha',
           placeholder: '••••••',
           isPasswordField: true,
+          onChanged: (value) =>
```

Comment:
Why setting it in onChanged? This way everytime the user press a key it will call the provider to update the email. Here is my suggestion:

1. Add an optional parameter to `text_input.dart`, the `TextFieldController? controller` and pass it as parameter to the `TextFormField` in it

2. Create a `TextFieldController` in the file `auth_model.dart` to each one of the inputs. It might be something like `TextEditingController emailController = TextEditingController();`, `TextEditingController passwordController = TextEditingController();`,

3. Pass the controllers as parameter to the inputs classes in `auth_page.dart`, like: 

```
AVAITextInput(
          label: 'E-mail institucional',
          placeholder: 'nome@ic.ufrj.br',
          allowNumbers: true,
          allowSpaces: false,
          controller: auth_provider.emailController
        ),
```

4. And then in the `AVAIButton` add a function that when the user presses the button it runs something like:

```
const emailValue = auth_provider.emailController.value.text;
const passwordValue= auth_provider.passwordValue.value.text;
auth_provider.setEmail(emailValue );
auth_provider.setPassword(passwordValue);
```